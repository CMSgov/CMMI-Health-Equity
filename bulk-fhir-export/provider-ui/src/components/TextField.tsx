import React from "react";
import { IconInfo } from "@trussworks/react-uswds";
import "../styles/text-field.css";
import "@trussworks/react-uswds/lib/uswds.css";
import "@trussworks/react-uswds/lib/index.css";
import { usePopperTooltip } from "react-popper-tooltip";
import "react-popper-tooltip/dist/styles.css";
import { ContextTooltip } from "./ContextTooltip";

type TextFieldProps = {
  value?: string;
  onChange?: Function;
  label?: string;
  name?: string;
  id?: string;
  error?: string;
  readOnly?: boolean;
  disabled?: boolean;
  tooltip?: string;
};

export default function TextField({
  value,
  onChange,
  label,
  id,
  name,
  error,
  readOnly,
  disabled,
  tooltip,
}: TextFieldProps): JSX.Element {
  const {
    getArrowProps,
    getTooltipProps,
    setTooltipRef,
    setTriggerRef,
    visible,
  } = usePopperTooltip({ placement: "top" });
  let errorEl =
    error == null ? null : (
      <span className="usa-error-message" id="input-error-message">
        {error}
      </span>
    );

  let labelEl =
    label == null ? null : (
      <label className="usa-label" htmlFor={name ?? ""}>
        {label}
        {tooltip && (
          <span
            ref={setTriggerRef}
            className="margin-left-1 position-relative top-05"
          >
            <IconInfo style={{ fill: "#0071BC" }} />
          </span>
        )}
        {tooltip && visible && (
          <ContextTooltip
            text={tooltip}
            setRef={setTooltipRef}
            getProps={getTooltipProps}
            getArrowProps={getArrowProps}
          />
        )}
      </label>
    );

  return (
    <React.Fragment>
      {labelEl}
      {errorEl}
      <input
        readOnly={readOnly ?? false}
        disabled={disabled ?? false}
        className="usa-input"
        id={id ?? ""}
        name={name ?? ""}
        type="text"
        value={value || ""}
        onChange={(e) => {
          if (onChange != null) onChange(e);
        }}
      />
    </React.Fragment>
  );
}
