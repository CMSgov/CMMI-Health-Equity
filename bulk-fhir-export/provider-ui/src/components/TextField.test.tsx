import React from 'react';
import renderer from 'react-test-renderer';
import TextField from './TextField';

test('Render TextField', () => {
  const component = renderer.create(
    <TextField label='test text field'/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
