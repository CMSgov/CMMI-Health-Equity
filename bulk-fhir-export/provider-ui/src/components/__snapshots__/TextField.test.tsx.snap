// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TextField 1`] = `
Array [
  <label
    className="usa-label"
    htmlFor=""
  >
    test text field
  </label>,
  <input
    className="usa-input"
    disabled={false}
    id=""
    name=""
    onChange={[Function]}
    readOnly={false}
    type="text"
    value=""
  />,
]
`;
